src/slow_hash.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use crate::errors::InternalPakeError;
+
+use generic_array::GenericArray;
+use scrypt::{scrypt, ScryptParams};
+use sha2::{Digest, Sha256};
+
+pub trait SlowHash {
+    fn hash(
+        input: GenericArray<u8, <Sha256 as Digest>::OutputSize>,
+    ) -> Result<Vec<u8>, InternalPakeError>;
+}
+
+pub struct NoOpHash;
+
+impl SlowHash for NoOpHash {
+    fn hash(
+        input: GenericArray<u8, <Sha256 as Digest>::OutputSize>,
+    ) -> Result<Vec<u8>, InternalPakeError> {
+        Ok(input.to_vec())
+    }
+}
+
+pub struct Scrypt;
```

Comment:
you could do an `impl SlowHash` for `ScryptParams` instead, which would incidentally let anybody that knows how to construct a set of ScryptParams configure this.